<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta name="generator" content="pandoc">
    <title>Finding Causative Mutations: Finding Causative Mutations With A Candidate SNP Approach</title>
    <link rel="shortcut icon" type="image/x-icon" href="/favicon.ico" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <link rel="stylesheet" type="text/css" href="css/bootstrap/bootstrap.css" />
    <link rel="stylesheet" type="text/css" href="css/bootstrap/bootstrap-theme.css" />
    <link rel="stylesheet" type="text/css" href="css/swc.css" />
    <!-- ><link rel="alternate" type="application/rss+xml" title="Software Carpentry Blog" href="http://software-carpentry.org/feed.xml"/> -->
    <meta charset="UTF-8" />
    <!-- HTML5 shim, for IE6-8 support of HTML5 elements -->
    <!--[if lt IE 9]>
      <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body class="lesson">
    <div class="container card">
      <div class="banner">
      <!--   <a href="http://software-carpentry.org" title="Software Carpentry">
          <img alt="Software Carpentry banner" src="img/software-carpentry-banner.png" />
      	</a> -->
      	<a href="http://usegalaxy.org" title="Galaxy Community Conference 2015 Training Course - Finding Causative Mutations Using A Candidate SNP Approach">
      	<img alt="Galaxy Logo" src="img/galaxy.png">	
      	<a href="gcc2015.tsl.ac.uk" title="Galaxy Community Conference 2015 @ TSL">
      	<img alt="GCC 2015" src="img/castle.png"> 
        </a>
        <span style="font-family:'Open Sans',sans-serif;font-size: 1.8em; font-weight: 400;"> GCC 2015 </span> <span><Finding Causative Mutations With A Candidate SNP approach</span>
      </div>
      <article>
      <div class="row">
        <div class="col-md-10 col-md-offset-1">
                    <a href="index.html"><h1 class="title">Finding Causative Mutations With A Candidate SNP Approach</h1></a>
          <h2 class="subtitle">Pre-Processing Sequence Data</h2>
          <aside class="callout panel panel-info">
<div class="panel-heading">
<h2 id="did-you-download-the-lesson-vm"><span class="glyphicon glyphicon-pushpin"></span>Did you download the lesson VM?</h2>
</div>
<div class="panel-body">
<p>You will need to already have some files downloaded.</p>
<ol style="list-style-type: decimal">
<li>VirtualBox software for your machine</li>
<li>The lesson Specific VM.</li>
</ol>
<p>Without these you are going to be in trouble, please make sure you downloaded them before the lesson.</p>
</div>
</aside>
<section class="objectives panel panel-warning">
<div class="panel-heading">
<h2 id="learning-objectives"><span class="glyphicon glyphicon-certificate"></span>Learning Objectives</h2>
</div>
<div class="panel-body">
<ul>
<li>Understanding Fastq</li>
<li>Inspecting and interpreting the quality of sequence data with FASTQC</li>
<li>Cleaning out sequence data with trimmomatic.</li>
</ul>
</div>
</section>
<p><a href="reference.html#Fastq">Fastq</a> is a typical sequence format generated by HTS machines. It contains four lines, a sequence ID, the sequence, the ID again and a messy looking quality string made up of characters, each of which represents the quality of the base above it. Here’s an example:</p>
<pre class="output"><code> @SEQ_ID
 GATTTGGGGTTCAAAGCAGTATCGATCAAATAGTAAATCCATTTGTTCAACTCACAGTTT
 +SEQ_ID
 !&#39;&#39;*((((***+))%%%++)(%%%%).1***-+*&#39;&#39;))**55CCF&gt;&gt;&gt;&gt;&gt;&gt;CCCCCCC65</code></pre>
<p>Each of the weird characters represents a number according to the <a href="reference.html#ASCII"><code>ASCII</code></a> look up table, where numbers are linked to characters, so <code>!</code> means <code>33</code> and <code>&quot;</code> means 34. These numbers are generally <a href="reference.html#Phred"><code>Phred</code></a> scores, which encode the likelihood of the base being wrong on a log scale.</p>
<p>We can use this quality information to assess how well our sequencing went. Along with sequence quality information we should also assess the composition of the sequence data. A program called <a href="reference.html#FastQC"><code>FastQC</code></a> is useful for this.</p>
<p>FastQC presents a range of plots and summary statistics, you need to provide it with Fastq data.</p>
<p>A typical output is like this:</p>
<div class="figure">
<img src="img/fastqc.png" alt="FastQC Summary Plot" />
<p class="caption">FastQC Summary Plot</p>
</div>
<p>This is the <code>per base sequence quality</code>. The plot shows along the x-axis the position in the read and for each position in the reads it shows a box-plot of all the quality scores at that position.</p>
<p>Noticeably in this figure, the box plots to the left have much higher and tightly grouped quality scores than those on the right. This is typical of Illumina machine sequence, the quality decreases the further you get along the read. As you can infer from the red region of the plot background, Phred scores less than 20 are generally not trusted.</p>
<aside class="callout panel panel-info">
<div class="panel-heading">
<h2 id="quality-score-encoding"><span class="glyphicon glyphicon-pushpin"></span>Quality score encoding</h2>
</div>
<div class="panel-body">
<p>Different sequencers use slightly different variations on the Phred score, this is usually called the <a href="reference.html#encoding">quality encoding</a>. Older Illumina pipelines encoded a score from -5 to 62 using ASCII characters 59 to 126, but nowadays most use Sanger encoding, which encodes a score from 0 to 93 using ASCII 33 to 126. Because of this discrepancy, it is necessary to sometimes be explicit about the sequence encoding in Galaxy. We do this by setting the data attributes of data files</p>
</div>
</aside>
<p>We may (or may not) decide that we need to get rid of the lower quality sequence.</p>
<p>At the individual sequence read level, we can discard entire sequences if part is too poor or trim the read leaving the good part alone. One system for doing this is <a href="reference.html#trimmomatic">Trimmomatic</a> which can perform a variety of trimming operations on sequence reads. It can remove parts of reads from the left or right sides up to quality thresholds - it uses a sliding window average, rather than just a harsh cut-off.</p>
<aside class="callout panel panel-info">
<div class="panel-heading">
<h2 id="sometimes-we-shouldnt-trim"><span class="glyphicon glyphicon-pushpin"></span>Sometimes we shouldn’t trim</h2>
</div>
<div class="panel-body">
<p>If we do carry out trimming, then we may end up with lots of reads of different lengths - this can be a problem for some aligners and downstream tools, so sometimes trimming isn’t the best strategy, we have to make context dependent decisions.</p>
</div>
</aside>
<p>At the sequence sample level (e.g, the read file level), we may discover that our read set is not good. Reports from FastQC like the <a href="reference.html#kmer"><code>kmer</code></a> content plot can show sequence problems, in this graph there is an over-representation of particular kmers at the start of the sequence that shouldn’t be there.</p>
<div class="figure">
<img src="img/kmer_content.png" alt="The first few bases here are significantly enriched, this can be due to sequence adapters (if they were used) but if not, then the sequence is likely not good, even if the quality scores are fine" />
<p class="caption">The first few bases here are significantly enriched, this can be due to sequence adapters (if they were used) but if not, then the sequence is likely not good, even if the quality scores are fine</p>
</div>
<aside class="callout panel panel-info">
<div class="panel-heading">
<h2 id="your-task-now-is-to-load-up-galaxy-and-run-some-reads-through-quality-control-and-trimming-prior-to-downstream-use."><span class="glyphicon glyphicon-pushpin"></span>Your task now is to load up Galaxy and run some reads through quality control and trimming, prior to downstream use.</h2>
</div>
<div class="panel-body">

</div>
</aside>
<aside class="callout panel panel-info">
<div class="panel-heading">
<h2 id="power-up-the-vm"><span class="glyphicon glyphicon-pushpin"></span>Power Up The VM</h2>
</div>
<div class="panel-body">
<ol style="list-style-type: decimal">
<li>Start <code>VirtualBox</code> by double-clicking it’s application icon.</li>
<li>Use <code>File</code> .. <code>Import Appliance</code> and select the <code>GCC2015-CandiSNP.ova</code> VM file.</li>
<li>Wait until it finishes importing and press the green <code>start</code> button</li>
<li>Login. The password is <code>12345</code></li>
</ol>
</div>
</aside>
<aside class="callout panel panel-info">
<div class="panel-heading">
<h2 id="connecting-to-the-internet"><span class="glyphicon glyphicon-pushpin"></span>Connecting to the internet</h2>
</div>
<div class="panel-body">
<p>You may need to change the VirtualBox network adapter if you get a report like:</p>
<pre class="error"><code>
en3 network adapter not found
</code></pre>
<p>Try using VirtualBox’s other network settings under <code>Devices</code> .. <code>Network</code> .. <code>Network Settings..</code>. Usually you should select, <code>NAT</code>. You can tell whether it’s working by using the web browser within the machine and navigating to an external site e.g <a href="http://bbc.co.uk" class="uri">http://bbc.co.uk</a></p>
</div>
</aside>
<aside class="callout panel panel-info">
<div class="panel-heading">
<h2 id="starting-galaxy"><span class="glyphicon glyphicon-pushpin"></span>Starting Galaxy</h2>
</div>
<div class="panel-body">
<p>You can either:</p>
<ol style="list-style-type: decimal">
<li><p>Double click the <code>Launch Galaxy</code> icon on the Desktop. You’ll need to be patient, Galaxy takes a little while to launch. (it should take about a minute to launch, so don’t panic if it doesn’t ping into life immediately) Or</p></li>
<li><p>Start a terminal <code>cd /var/git/galaxy</code>, <code>sh run.sh</code>, navigate to <code>localhost:8080</code> with the web browser.</p></li>
</ol>
</div>
</aside>
<section class="challenge panel panel-success">
<div class="panel-heading">
<h2 id="run-fastqc"><span class="glyphicon glyphicon-pencil"></span>1. Run FastQC</h2>
</div>
<div class="panel-body">
<p>Use the reads in the <code>QC</code> <code>Shared Data</code> library. You will find the <code>FastQC</code> tool in the tool list under <code>HTS QC</code>. The reads are single ended from a mutagenised <em>Arabidopsis thaliana</em> (a model plant). They are Illumina <a href="reference.html#whole_genome_shotgun">Whole Genome Shotgun reads</a> so don’t have any multiplex adapters and the plants are grown in sterile culture so we aren’t expecting contamination.</p>
<ol style="list-style-type: decimal">
<li>How many reads are you using?</li>
<li>What sort of output files do you get from FastQC?</li>
<li>What should you do with these files? Do they represent a scientific control that could be published?</li>
</ol>
</div>
</section>
<section class="challenge panel panel-success">
<div class="panel-heading">
<h2 id="interpret-sequence-quality"><span class="glyphicon glyphicon-pencil"></span>2. Interpret Sequence Quality</h2>
</div>
<div class="panel-body">
<ol style="list-style-type: decimal">
<li>Is there any evidence of contamination? Which report tells you?</li>
<li>If there is, which sequence is contaminating?</li>
<li>Which of the four files is the best?</li>
</ol>
</div>
</section>
<section class="challenge panel panel-success">
<div class="panel-heading">
<h2 id="clean-up-poor-quality-sequence"><span class="glyphicon glyphicon-pencil"></span>3. Clean Up Poor Quality Sequence</h2>
</div>
<div class="panel-body">
<p>Use the <code>Trimmomatic</code> tool in <code>HTS QC</code>. Select an appropriate <code>Average quality required</code></p>
<ol style="list-style-type: decimal">
<li>Find a trimming strategy to get rid of contamination you observed in section <code>2.1</code>?</li>
<li>Which trimming strategy improves the low quality reads</li>
<li>Did it make any difference?</li>
<li>How could you filter on size? If you needed to pass only good quality, full length sequences to the next step?</li>
</ol>
</div>
</section>
<h2 id="to-do" class="todo">TO DO</h2>
<p>Set filename and url for VMs</p>
        </div>
      </div>
      </article>
      <div class="footer">
        <!-- <a class="label swc-blue-bg" href="http://software-carpentry.org">Software Carpentry</a> -->
        <a class="label swc-blue-bg" href="https://github.com/swcarpentry/lesson-template">Source</a>
        <!-- <a class="label swc-blue-bg" href="mailto:admin@software-carpentry.org">Contact</a> -->
        <a class="label swc-blue-bg" href="mailto:dan.maclean@tsl.ac.uk">Contact</a>
        <a class="label swc-blue-bg" href="LICENSE.html">License</a>
      </div>
    </div>
    <!-- Javascript placed at the end of the document so the pages load faster -->
    <script src="http://software-carpentry.org/v5/js/jquery-1.9.1.min.js"></script>
    <script src="css/bootstrap/bootstrap-js/bootstrap.js"></script>
  </body>
</html>
