<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta name="generator" content="pandoc">
    <title>Finding Causative Mutations: Finding Causative Mutations With A Candidate SNP Approach</title>
    <link rel="shortcut icon" type="image/x-icon" href="/favicon.ico" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <link rel="stylesheet" type="text/css" href="css/bootstrap/bootstrap.css" />
    <link rel="stylesheet" type="text/css" href="css/bootstrap/bootstrap-theme.css" />
    <link rel="stylesheet" type="text/css" href="css/swc.css" />
    <!-- ><link rel="alternate" type="application/rss+xml" title="Software Carpentry Blog" href="http://software-carpentry.org/feed.xml"/> -->
    <meta charset="UTF-8" />
    <!-- HTML5 shim, for IE6-8 support of HTML5 elements -->
    <!--[if lt IE 9]>
      <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body class="lesson">
    <div class="container card">
      <div class="banner">
      <!--   <a href="http://software-carpentry.org" title="Software Carpentry">
          <img alt="Software Carpentry banner" src="img/software-carpentry-banner.png" />
      	</a> -->
      	<a href="http://summerschool.tsl.ac.uk" title="TSL Summer School 2015 - Finding Causative Mutations Using A Candidate SNP Approach">
      	<img alt="TSL Logo" src="img/logo.png">	
      	<a href="summerschool.tsl.ac.uk" title="Summer School 2015 @ TSL">
        </a>
        <span style="font-family:'Open Sans',sans-serif;font-size: 1.8em; font-weight: 400;">  </span> <span>Finding Causative Mutations With A Candidate SNP approach</span>
      </div>
      <article>
      <div class="row">
        <div class="col-md-10 col-md-offset-1">
                    <a href="index.html"><h1 class="title">Finding Causative Mutations With A Candidate SNP Approach</h1></a>
          <h2 class="subtitle">Visualising SNPs To Find Candidates</h2>
          <section class="objectives panel panel-warning">
<div class="panel-heading">
<h2 id="learning-objectives"><span class="glyphicon glyphicon-certificate"></span>Learning Objectives</h2>
</div>
<div class="panel-body">
<ul>
<li>Understand CandiSNP output</li>
<li>Plotting frequencies of SNPs of different types across the chromosome</li>
</ul>
</div>
</section>
<p>Once we have a list of SNPs that we are happy with and have annotated them with SNPEff, there are a couple of approaches we can take to start finding candidates that may be our causative mutation.</p>
<p>The approach we take will depend on the genetic background. As we discussed at the start, we are generally looking for a region of high homozygous SNPs, but the frequency of other SNPs will depend on the cross. A wide cross from a fairly distant relative (like a different strain or ecotype) as is commonly used in genetic mapping strategies will allow us to make use of the heterozygous SNPs as a control.</p>
<p>Fast interactive visualisations are a great help in finding the recombinant region and narrowing the candidates. One tool that allows us to do this is CandiSNP. CandiSNP is a JavaScript visualisation package that allows interactive filtering and highlighting of SNPs across whole chromosomes (DISCLAIMER: My group wrote this!)</p>
<p>It allows you to look at the SNPs like this: <img src="img/candisnp.png" alt="CandiSNP interface" /></p>
<p>and apply filters to narrow down the region and candidates so you see this:</p>
<div class="figure">
<img src="img/candisnp_after.png" alt="CandiSNP after filtering - the region of the high red spot density is the recombinant one" />
<p class="caption">CandiSNP after filtering - the region of the high red spot density is the recombinant one</p>
</div>
<p>CandiSNP takes an annotated VCF file as input.</p>
<p>Statistical methods are useful when the number of SNPs generated is so large that you can’t visualise them all at the same time.</p>
<p>Density plots like this one (which is of the same data as the CandiSNP):</p>
<div class="figure">
<img src="img/density.png" alt="Density plot of Homozygous, Heterozygous SNP density and the ratio of Hom/Het SNPS in slifing windows" />
<p class="caption">Density plot of Homozygous, Heterozygous SNP density and the ratio of Hom/Het SNPS in slifing windows</p>
</div>
<p>help us to see the rough patterns in a similar way. The homozygous and heterozygous show an increase in the SNP-rich centromeric region which biases the data and an overall decrease at the far right of the chromosome, but the enriched region is visible in the high ratio at about 17Mbp as in the CandiSNP output. These kinds of plots can be generated with Galaxy’s plotting tools.</p>
<aside class="callout panel panel-info">
<div class="panel-heading">
<h2 id="centromeres"><span class="glyphicon glyphicon-pushpin"></span>Centromeres</h2>
</div>
<div class="panel-body">
<p>Centromeres are a real problem with these sorts of analysis. They are so SNP rich that they swamp analysis and visualisations. It helps to just screen them out from the analysis. CandiSNP will let you turn off centromeres associated SNPs, Galaxy tools can also help you filter them out.</p>
</div>
</aside>
<aside class="callout panel panel-info">
<div class="panel-heading">
<h2 id="snp-deletion---fewer-are-better"><span class="glyphicon glyphicon-pushpin"></span>SNP Deletion - Fewer are better</h2>
</div>
<div class="panel-body">
<p>Perhaps counter-intuitively, getting fewer SNPs is often better in these approaches. A common source of confounding SNPs is from the parental line itself. All individuals of any species have differences in the genomes from the references we use to call SNPs, and some (perhaps many) of these will be shared between the parent used to generate the mutants and the mutants. By sequencing the parent line and calling SNPs between it and the reference genome, you get a list of parental SNPs that you can often delete straight out of the mutant as being non-causative.</p>
</div>
</aside>
<aside class="callout panel panel-info">
<div class="panel-heading">
<h2 id="c.elegans-and-cloudmap"><span class="glyphicon glyphicon-pushpin"></span><em>C.elegans</em> and CloudMap</h2>
</div>
<div class="panel-body">
<p>Galaxy Team members have also produced a tool for doing this sort of thing. It’s called CloudMap and it provides visualisation and plenty of options for finding non-recombinant regions in the backcrossed lines. It helpfully allows you to filter out SNPs that could be candidates, such as EMS induced ones. As far as I can tell, the version available as a Galaxy tool works on <em>C.elegans</em>.</p>
</div>
</aside>
<section class="challenge panel panel-success">
<div class="panel-heading">
<h2 id="analyse-snp-data-with-candisnp"><span class="glyphicon glyphicon-pencil"></span>Analyse SNP data with CandiSNP</h2>
</div>
<div class="panel-body">
<p>You have some whole genome <em>Arabidopsis</em> SNP data annotated with SNPEff in the shared data library <code>Visualisation</code>, the VCF file <code>filtered_snps.vcf</code>. Use this in the <code>candisnp</code> tool in the <code>CandiSNP</code> tool set. This data set is a real one and we know exactly where the mutation is because we’ve sequenced it, so there is a <em>right</em> answer. Use the sliders and filter tools to find a region enriched in homozygous candidate SNPs.</p>
<ol style="list-style-type: decimal">
<li>Can you come up with candidate regions / genes for the causative mutation?</li>
<li>Which is more useful, filtering or colouring?</li>
<li>How much extra information does knowing the genes give? In this specific training case and in a real case where you might know something about the biology already.</li>
</ol>
</div>
</section>
<section class="challenge panel panel-success">
<div class="panel-heading">
<h2 id="generate-density-plots-of-different-snp-classes"><span class="glyphicon glyphicon-pencil"></span>Generate density plots of different SNP classes</h2>
</div>
<div class="panel-body">
<p>Standard Galaxy tools can generate histograms of data. However the data needs to be in tabular format, not VCF. Here’s a little recipe for going from VCF to a table that is useful.</p>
<ol style="list-style-type: decimal">
<li>To make a tabular file, use the <code>Text Manipulation .. Cut Columns Tool</code>. Cut out columns <code>c1,c2,c3,c4,c8,c9</code>.</li>
<li>To strip text from the AF field and just leave the numbers, use the <code>Text Manipulation .. Trim Leading</code> on <code>column 5</code>, trim to position <code>4</code> and set to <code>ignore</code> <code>#</code>.</li>
<li>To get the homozygous SNPs, use the <code>Filter and Sort .. Filter data on any column</code> tool. Filter on <code>c5&gt;=0.75</code> (or whatever seems sensible to you).</li>
<li>To get the heterozygous SNPs, do step <code>3</code> again but filter on <code>c5&lt;0.75</code>.</li>
</ol>
<p>This will leave you with two files on which to carry out the remaining steps.</p>
<ol start="5" style="list-style-type: decimal">
<li>To split the files into single chromosome files use <code>Text Manipulation .. Split file according to value of a column</code> use <code>column 1</code> (the chromosome column)</li>
<li>For each resulting file you can use <code>Plotting .. Histogram</code> to make the histograms. It is useful to add the density plot and to use around 150 breaks.</li>
</ol>
<p>With a bit more work you can combine the different plots into one large one for easier comparison, and you can use sliding window tools to calculate the ratio of Hom/Het SNPs across the chromosomes.</p>
</div>
</section>
<section class="challenge panel panel-success">
<div class="panel-heading">
<h2 id="comparing-density-plots"><span class="glyphicon glyphicon-pencil"></span>Comparing density plots</h2>
</div>
<div class="panel-body">
<ol style="list-style-type: decimal">
<li>Use the numerous density plots you made to compare the likely positions of the causative SNPs</li>
<li>Can you narrow down the area sufficiently to examine the text list in more detail?</li>
<li>What further filtering could you do to make the plots more effective?</li>
</ol>
</div>
</section>
        </div>
      </div>
      </article>
      <div class="footer">
        <!-- <a class="label swc-blue-bg" href="http://software-carpentry.org">Software Carpentry</a> -->
        <a class="label swc-blue-bg" href="https://github.com/swcarpentry/lesson-template">Source</a>
        <!-- <a class="label swc-blue-bg" href="mailto:admin@software-carpentry.org">Contact</a> -->
        <a class="label swc-blue-bg" href="mailto:dan.maclean@tsl.ac.uk">Contact</a>
        <a class="label swc-blue-bg" href="LICENSE.html">License</a>
      </div>
    </div>
    <!-- Javascript placed at the end of the document so the pages load faster -->
    <script src="http://software-carpentry.org/v5/js/jquery-1.9.1.min.js"></script>
    <script src="css/bootstrap/bootstrap-js/bootstrap.js"></script>
  </body>
</html>
