---
layout: lesson
title: Assembling a draft genome in the Galaxy Interface
---

In this lesson we will look at the powerful Galaxy Workflow Environment in which we will run an assembly of some High Throughput Illumina reads.

Sadly we don't have the scope or time to do an assembly of a whole genome - the algorithms take quite a while to run, so we'll do a toy example - selected reads from _real data_ that will at least complete in a sensible time. 

We'll look at what it takes to work out running an assembly and doing the proper quality control to make sure that it is of good quality once it is done.

>## Prerequisites {.prereq}
> This course walks you gently through the Galaxy Interface and gives you opportunity to become familiar with that as a way of running bioinformatics tools. There are no formal pre-requisites for this, as long as you've used a web-browser and mouse, you will be fine!
> 

> ## Getting ready {.getready}
>
> You can find the TSL Galaxy Server at [http://galaxy.tsl.ac.uk](http://galaxy.tsl.ac.uk)
>
> Log-in to the server using the username and passwords provided at the front of the class.
>

## Topics

1.  [Why genomics?](motivation.html)
2.  [Learning the Galaxy Interface](01-galaxy.html)
3.  [Pre-Processing Sequence Data](02-preprocessing.html)
4.  [Assembling and assessing assemblies](03-assembly.html)

## Other Resources

*   [Reference](reference.html)