<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta name="generator" content="pandoc">
    <title>Finding Effector Genes and Proteins</title>
    <link rel="shortcut icon" type="image/x-icon" href="/favicon.ico" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <link rel="stylesheet" type="text/css" href="css/bootstrap/bootstrap.css" />
    <link rel="stylesheet" type="text/css" href="css/bootstrap/bootstrap-theme.css" />
    <link rel="stylesheet" type="text/css" href="css/swc.css" />
    <!-- ><link rel="alternate" type="application/rss+xml" title="Software Carpentry Blog" href="http://software-carpentry.org/feed.xml"/> -->
    <meta charset="UTF-8" />
    <!-- HTML5 shim, for IE6-8 support of HTML5 elements -->
    <!--[if lt IE 9]>
      <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body class="lesson">
    <div class="container card">
      <div class="banner">
      <!--   <a href="http://software-carpentry.org" title="Software Carpentry">
          <img alt="Software Carpentry banner" src="img/software-carpentry-banner.png" />
      	</a> -->
      	<a href="http://summerschool.tsl.ac.uk" title="TSL Summer School 2015 - Finding Causative Mutations Using A Candidate SNP Approach">
      	<img alt="Galaxy Logo" src="img/logo.png">	
      	<a href="summerschool.tsl.ac.uk" title="Summer School 2015 @ TSL">
        </a>
        <span style="font-family:'Open Sans',sans-serif;font-size: 1.8em; font-weight: 400;">  </span> <span>Finding Causative Mutations With A Candidate SNP approach</span>
      </div>
      <article>
      <div class="row">
        <div class="col-md-10 col-md-offset-1">
                    <a href="index.html"><h1 class="title">Finding Effector Genes and Proteins</h1></a>
          <h2 class="subtitle">Finding Signal Peptides</h2>
          <section class="objectives panel panel-warning">
<div class="panel-heading">
<h2 id="learning-objectives"><span class="glyphicon glyphicon-certificate"></span>Learning Objectives</h2>
</div>
<div class="panel-body">
<ul>
<li>Understand Classification</li>
<li>Understand what makes good positive and negative controls</li>
<li>Run SignalP and interpret the output</li>
</ul>
</div>
</section>
<p>Classification is the process of taking an object and deciding whether it falls into a particular group. In this case we will be deciding whether a protein is an effector or not. Classification at its simplest requires some sort of baseline against which to compare our candidate and make a decision so the first step of classification is to create the classifier. Often these classifiers will be data structures embodied in computer programs. The sophistication of these programs varies and their contribution to the accuracy of classification varies too. Most importantly when the program contains its own model we must be aware of how that model can vary from what it is we are looking for. Therefore we need to always apply the best scientific principles of control and criticism when carrying out these analyses, these tools will all give spurious results if we try to use them as a <code>black-box</code>.</p>
<p>Appropriate controls are vitally important. In this tutorial you<code>ll notice that the</code>right` control is probably a matter of opinion and that the interpretation and results of our analysis can vary hugely dependent on what we are using. As far as possible it is necessary to get a control or control set that has been shown experimentally to be of the class that we are interested in.</p>
<p>Protein secretion is a primary mechanism in the pathogenesis of many prokaryotic and eukaryotic microorganisms. The effector proteins are secreted from the cytosol of the pathogen into the host whereby they cause disease. The molecular signal that marks a protein for export is called the signal peptide and is typically found at the N-terminal of the protein. Thus candidates can be screened by identifying a signal peptide. Signal peptide programs are numerous, SignalP is one of that is frequently used.</p>
<aside class="callout panel panel-info">
<div class="panel-heading">
<h2 id="hidden-markov-models"><span class="glyphicon glyphicon-pushpin"></span>Hidden Markov Models</h2>
</div>
<div class="panel-body">
<p>Hidden Markov Models (HMMs) are complex statistical tools that work by calculating the probability of moving from one state to another based on a background model, that is to say from moving from an <code>L</code> amino acid to a <code>D</code> given the relative proportions of <code>L</code>s and <code>D</code>s in that point in the set of sequences from which the model was derived. The canonical tool for using and defining HMMs is HMMER. The process of defining a HMM requires a careful alignment of a number of positively identified proteins. HMMs tend to be used for defining short protein domains or motifs. Many methods dealing with sequence use HMMs in one way or another.</p>
</div>
</aside>
<p>From the author`s description <sub>~</sub> SignalP is a program that we can use to find potential peptide signals. SignalP server predicts the presence and location of signal peptide cleavage sites in amino acid sequences from different organisms: Gram-positive prokaryotes, Gram-negative prokaryotes, and eukaryotes. The method incorporates a prediction of cleavage sites and a signal peptide/non-signal peptide prediction based on a combination of several artificial neural networks and hidden Markov models. <sub>~</sub></p>
<p>It is therefore a complex heuristic and its internal models of what constitute a signal peptide may in fact be very different from those that you are interested in, which means that we must examine the output very carefully.</p>
<section class="challenge panel panel-success">
<div class="panel-heading">
<h2 id="run-signalp-3.0"><span class="glyphicon glyphicon-pencil"></span>Run SignalP 3.0</h2>
</div>
<div class="panel-body">
<ul>
<li>Start a new history and from the <code>Summer School</code> data library, load in the data in <code>Effector Classification</code></li>
<li>Find the SignalP 3.0 tool under <code>Protein Analysis</code></li>
<li>Select your <code>unclassified.fa</code> file as the sequences in <code>Fasta file of protein sequences</code></li>
<li>Select Eukaryote</li>
<li>Select 70 for <code>Truncate Sequences</code></li>
<li>Click <code>Execute</code></li>
<li>You`ll get a big table as a result, you might find it easier to examine if you copy and paste into Excel.</li>
</ul>
</div>
</section>
<p>As you will see from the output SignalP makes a definitive prediction about whether a protein carries a signal peptide or not, however as this is based on its internal model we need to compare some known proteins. Look at the scores and read the SignalP output description in the Galaxy tool window to get a feel for what it is telling you. Here<code>s SignalP</code>s own description of the scores:</p>
<aside class="callout panel panel-info">
<div class="panel-heading">
<h2 id="signalp-scores"><span class="glyphicon glyphicon-pushpin"></span>SignalP scores</h2>
</div>
<div class="panel-body">
<p>The graphical output from SignalP (neural network) comprises three different scores, C, S and Y. Two additional scores are reported in the SignalP output, namely the S-mean and the D-score, but these are only reported as numerical values.</p>
<ul>
<li><p>The <code>S-score</code> for the signal peptide prediction is reported for every single amino acid position in the submitted sequence, with high scores indicating that the corresponding amino acid is part of a signal peptide, and low scores indicating that the amino acid is part of a mature protein.</p></li>
<li><p>The <code>C-score</code> is the <code>cleavage site</code> score. For each position in the submitted sequence, a C-score is reported, which should only be significantly high at the cleavage site. Confusion is often seen with the position numbering of the cleavage site. When a cleavage site position is referred to by a single number, the number indicates the first residue in the mature protein, meaning that a reported cleavage site between amino acid 26-27 corresponds to that the mature protein starts at (and include) position 27.</p></li>
<li><p>The <code>Y-max</code> is a derivative of the C-score combined with the S-score resulting in a better cleavage site prediction than the raw C-score alone. The cleavage site is assigned from the Y-score where the slope of the S-score is steep and a significant C-score is found.</p></li>
</ul>
<p>The <code>S-mean</code> is the average of the S-score, ranging from the N-terminal amino acid to the amino acid assigned with the highest Y-max score, thus the S-mean score is calculated for the length of the predicted signal peptide. The S-mean score was in SignalP version 2.0 used as the criteria for discrimination of secretory and non-secretory proteins.</p>
<p>The <code>D-score</code> is introduced in SignalP version 3.0 and is implemented as a weighted average of the S-mean and the Y-max scores. The score shows superior discrimination performance of secretory and non-secretory proteins to that of the S-mean score which was used in SignalP version 1 and 2.</p>
<p>For non-secretory proteins all the scores represented in the SignalP output should ideally be very low.</p>
</div>
</aside>
<div class="figure">
<img src="img/signalp.png" alt="SignalP output - note how the scores change at the cleavage site - at position 30." />
<p class="caption">SignalP output - note how the scores change at the cleavage site - at position 30.</p>
</div>
<section class="challenge panel panel-success">
<div class="panel-heading">
<h2 id="getting-a-good-result-from-signalp"><span class="glyphicon glyphicon-pencil"></span>Getting a good result from SignalP</h2>
</div>
<div class="panel-body">
<ul>
<li><p>Run SignalP on the <code>avr_proteins.fa</code> file</p></li>
<li>Compare the values of the scores, in particular the D scores, for the <code>avr_proteins.fa</code> and your unclassified proteins.</li>
<li><p>Can you come up with a convincing threshold that allows you to discriminate the candidate secreted peptides better than the raw SignalP output? Sketch out your ideas for the comparison.</p></li>
</ul>
</div>
</section>
<aside class="callout panel panel-info">
<div class="panel-heading">
<h2 id="manipulating-text-and-tables-in-galaxy"><span class="glyphicon glyphicon-pushpin"></span>Manipulating text and tables in Galaxy</h2>
</div>
<div class="panel-body">
<p>Galaxy is designed to work with text and tabular data, a lot of its tools perform useful actions on these types of files. In particular the tools in the categories below:</p>
<ul>
<li>Text Manipulation</li>
<li>Filter and Sort</li>
<li>Join, Subtract and Group</li>
</ul>
<p>are all very useful for making subsets of results and combining them with others. We won`t go through them in too much detail in this document, they are pretty self-explanatory, but they are useful if you want to automate the challenge above.</p>
</div>
</aside>
        </div>
      </div>
      </article>
      <div class="footer">
        <!-- <a class="label swc-blue-bg" href="http://software-carpentry.org">Software Carpentry</a> -->
        <a class="label swc-blue-bg" href="https://github.com/swcarpentry/lesson-template">Source</a>
        <!-- <a class="label swc-blue-bg" href="mailto:admin@software-carpentry.org">Contact</a> -->
        <a class="label swc-blue-bg" href="mailto:dan.maclean@tsl.ac.uk">Contact</a>
        <a class="label swc-blue-bg" href="LICENSE.html">License</a>
      </div>
    </div>
    <!-- Javascript placed at the end of the document so the pages load faster -->
    <script src="http://software-carpentry.org/v5/js/jquery-1.9.1.min.js"></script>
    <script src="css/bootstrap/bootstrap-js/bootstrap.js"></script>
  </body>
</html>
